### DISCUSSION TITLE
How can I fix the issue of GitHub showing an outdated or incorrect file version after pushing a commit?

### DISCUSSION THREAD

**Discussion by anotheradamibnu** [community/community#147605] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hi!
I’ve been encountering an issue where, after committing changes to my local Git repository and pushing them to GitHub, the website continues to display an outdated or incorrect version of a file. Even after multiple pushes, GitHub still shows the previous content, not the updated file.
How can I troubleshoot and resolve this issue?

**Comment by Adamibnualfatahh** [comment, marked as answer]
To fix the issue of GitHub showing an outdated file version:

1. **Ensure Local Changes are Pushed**: Run `git status`, commit with `git commit -m "message"`, and push using `git push origin main`.

2. **Check the Correct Branch**: Make sure you're viewing the right branch on GitHub.

3. **Force a Refresh**: Make a small change to the file and push again, e.g., `echo " " >> filename`.

4. **Check Commit History**: Confirm that your commit is registered both locally (`git log`) and on GitHub.

5. **Resolve Merge Conflicts**: If any, resolve conflicts with `git diff` and push the changes.

6. **Check File Case Sensitivity**: Ensure filenames' case matches on both your local and remote repositories.

7. **Force Push (if necessary)**: If all else fails, use `git push origin main --force` to overwrite the remote file.

**Reply by anotheradamibnu** [reply]
Thanks!!!

**Reply by Adamibnualfatahh** [reply]
You're welcome!

**Comment by phanbamanhpro** [comment]
Có vẻ hay đấy

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @anotheradamibnu!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Repositories category. I've gone ahead and moved it for you. Good luck!
